### DISCUSSION TITLE
Network change list

### DISCUSSION THREAD

**Discussion by antoniokranjcina** [android/nowinandroid#442] [Q&A] [answered]
Hi,

I have noticed that all three external models (`Topic`, `Author`, `NewsResource`) have their `ChangeList` objects which are fetched from network and then saved in local datastore in order to sync offline and network data through `Syncable`. My question here is, how should `syncWith()` work if there is no ChangeList objects from network datasource? How should `syncWith` be adapted?

**Comment by tunjid** [comment, marked as answer]
Hi there.

For synchronization to work, there needs to be som sort of token to let the app know how valid its data is relative to the network. If there are no tokens (in NiA, this is the change list), consider using a [pull based](https://developer.android.com/topic/architecture/data-layer/offline-first#pull-based-synchronization) approach.

In this case you may load the same data multiple times over, and there most likely won't be a `syncWith()` method. You VM may just call a `refresh()` method that goes to fetch data again. See an example of this in the [architecture samples repo](https://github.com/android/architecture-samples/blob/03d9af0b6a1ab82e450ca7c45f973a119412a9ab/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/DefaultTaskRepository.kt#L157).
